api/common/sharing/utils.go

```diff
@@ -37,13 +37,8 @@ func ExtractReferencedInstanceID(req *web.Request, repository storage.Repository
 			return "", err
 		}
 	} else {
-		sharedInstances, err := getSharedInstances(ctx, repository, referencePlan.ServiceOfferingID, tenantIdentifier, getTenantId())
-		if err != nil {
-			log.C(ctx).Errorf("Failed to retrieve shared instances: %s", err)
-			return "", err
-		}
-
-		referencedInstance, err = filterInstancesBySelectors(ctx, repository, sharedInstances, parameters, smaap, referencePlan.ServiceOfferingID)
+		selectors := parameters[instance_sharing.SelectorsKey].Map()
+		referencedInstance, err = filterInstancesBySelectors(ctx, repository, referencePlan.ServiceOfferingID, tenantIdentifier, getTenantId(), selectors, smaap)
 		if err != nil {
 			log.C(ctx).Errorf("Failed to filter instances by selectors: %s", err)
```

Comment:
In case of error, it will print the address of the error and the its details. 
You have 2 options use
1.  err.Error() - then it will give you the string 
2.  use %v in format

Reply:
thanks, using err.Error() instead.